dts/bindings/base/pm.yaml

```diff
@@ -30,3 +30,9 @@ properties:
     description: |
       Automatically configure the device for runtime power management after the
       init function runs.
+
+  reinit-power-states:
+    type: phandles
+    description: |
+      System power states in which a peripheral's configuration is lost, and
+      must be reinitialized to ensure correct operation.
```

Comment:
this has nothing to do with hw description, so it should use the `zephyr,` prefix.

Reply:
When it describes a (System PM) state in which the hardware changes, wouldn't that classify it as hardware description?

Reply:
By looking at e.g. https://www.st.com/resource/en/product_training/STM32L4_System_Power.pdf, I see this property more of a software-driving property than something that precisely describes thee hardware.

Reply:
I agree in this case this is really a hw description, as it can be found in ref man
![wl_table](https://github.com/zephyrproject-rtos/zephyr/assets/18574308/3b7842c3-c89d-402a-9bfa-d9a7e493660d)

Reply:
> I agree in this case this is really a hw description, as it can be found in ref man ![wl_table](https://user-images.githubusercontent.com/18574308/254269147-3b7842c3-c89d-402a-9bfa-d9a7e493660d.png)

I don't see LP states applying in the same way to all peripherals, so there seems to be a relationship between LP states and peripherals that this property doesn't capture.

Reply:
> I don't see LP states applying in the same way to all peripherals, so there seems to be a relationship between LP states and peripherals that this property doesn't capture.

That's why the property is intended to be used per-peripheral, e.g.

```c
		i2c2: i2c@40005800 {
                        ....
			reinit-power-states = <&stop2>;
                        ....
		};
```

Do you mean the wording or description of it doesn't capture the relationship? I could reword it if you think it's not clear enough.

Reply:
Note that the **Y**, **O**, and **R** classifications in the table above all mean the same thing in terms of peripheral configuration retention, if that is what you're touching on with your comment.
- **Y**: enabled, peripheral is operational during LP state
- **O**: optionally enabled (via config register bit) during LP state, but register/clock configuration is always retained
- **R**: register/clock configuration retained, though peripheral is unavailable during LP state

Only the lack of any classifier (denoted with a **-**) calls for peripheral reinit.
Adding a separate property to enable peripherals in LP states with an **O** would be useful, but is I believe out-of-scope for this PR's functionality.
Also note that the individual peripheral config bit to enable it applies to *all* LP states that have said **O** classifier, there's no option to enable for instance USART1 in STOP0, but not in STOP1, unless this is added to be done manually by System PM logic for every state exit/enter; I personally don't see a use-case for this.

Reply:
This is a hw property for me as well.